python/spacetime/store.py

```diff
@@ -1,478 +1,119 @@
-﻿'''
+'''
 Created on Apr 19, 2016
 
 @author: Rohan Achar
 '''
-
-from flask import Flask, request
-from flask.helpers import make_response
-from flask_restful import Api, Resource, reqparse
-from common.recursive_dictionary import RecursiveDictionary
-from common.converter import create_complex_obj, create_jsondict
-from datamodel.all import DATAMODEL_TYPES
-import pcc
 import json
-import os
-import sys
-import uuid
-import logging
-from threading import RLock
-
-from cache import Cache
-
-def calc_basecls2derived(name2class):
-    result = {}
-    for tp in DATAMODEL_TYPES:
-        if hasattr(tp, "__PCC_BASE_TYPE__") and tp.__PCC_BASE_TYPE__:
-            for base in tp.mro():
-                if (base != tp.Class()
-                        and base.__name__ in name2class
-                        and hasattr(name2class[base.__name__], "__PCC_BASE_TYPE__")
-                        and name2class[base.__name__].__PCC_BASE_TYPE__):
-                    result.setdefault(name2class[base.__name__], []).append(tp)
-    #                print base.__name__, tp.__name__
-    #print [(tp.__realname__, [der.__realname__ for der in result[tp]]) for tp in result]
-    return result
-
-
-
-# not active object.
-# just stores the basic sets.
-# requires set types that it has to store.
-# must accept changes.
-# must return object when asked for it.
-class store(object):
-    def __init__(self):
-        self.logger = logging.getLogger(__name__)
-        # type -> {id : object} object is just json style recursive dictionary.
-        # Onus on the client side to make objects
-        self.__data = RecursiveDictionary()
-        self.__base2derived = calc_basecls2derived(
-                    dict([(tp.__realname__, tp) for tp in DATAMODEL_TYPES]))
+from time import sleep
+from pcc.recursive_dictionary import RecursiveDictionary
+from datamodel.all import DATAMODEL_TYPES
+from pcc.dataframe import dataframe, DataframeModes
+
+class Modes(object):
+    Producing = "producing"
```

Comment:
I can't make them int. They happen to be keys of the json used to register the app. But it is a one time thing per app. Shouldn't be too bad.